sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/SubstituteUnresolvedOrdinals.scala

```diff
@@ -27,22 +27,21 @@ import org.apache.spark.sql.catalyst.trees.CurrentOrigin.withOrigin
 /**
  * Replaces ordinal in 'order by' or 'group by' with UnresolvedOrdinal expression.
  */
-class UnresolvedOrdinalSubstitution(conf: CatalystConf) extends Rule[LogicalPlan] {
-  private def isIntegerLiteral(sorter: Expression) = IntegerIndex.unapply(sorter).nonEmpty
+class SubstituteUnresolvedOrdinals(conf: CatalystConf) extends Rule[LogicalPlan] {
+  private def isIntLiteral(sorter: Expression) = IntegerIndex.unapply(sorter).nonEmpty
```

Comment:
as we are cleaning up the code, shall we also remove `IntegerIndex`? It became unnecessary after we make `-1` a literal instead of `UnaryMinus`.